app/src/main/kotlin/com/wire/android/ui/home/appLock/LockCodeTimeManager.kt

```diff
@@ -77,13 +79,26 @@ class LockCodeTimeManager @Inject constructor(
                     when {
                         appLockConfig is AppLockConfig.Disabled -> flowOf(false)
 
-                        !isInForeground && !isLockedFlow.value -> flow {
+                        !isInForeground && !isLockedFlow.value && appLockConfig is AppLockConfig.Enabled -> flow {
                             appLogger.i("$TAG lock is enabled and app in the background, lock count started")
+                            if (lockTimeoutStarted == null) {
+                                lockTimeoutStarted = currentTimestamp()
+                            }
                             delay(appLockConfig.timeout.inWholeMilliseconds)
-                            appLogger.i("$TAG lock count ended, app state should be locked if passcode is set")
+                            appLogger.i("$TAG lock count ended while app in the background, app state should be locked")
+                            emit(true)
```

Comment:
Do we still need to make this delay ? the app will check using timestamp after it

Reply:
I didn't want to change too much and potentially break the current logic which emits this change after the delay, instead I just wanted to add another solution with this timestamp as a backup in case this delay doesn't work as expected. With delay we can get this information more quickly and then there will be no need to check the timestamp when launching the app and that will show app lock screen instantly without the need to make any calculations.